diff --git a/src/googletest/cmake/internal_utils.cmake b/src/googletest/cmake/internal_utils.cmake
index 2f70f0b..51c6dfe 100644
--- a/src/googletest/cmake/internal_utils.cmake
+++ b/src/googletest/cmake/internal_utils.cmake
@@ -282,11 +282,11 @@ function(py_test name)
           add_test(NAME ${name}
             COMMAND powershell -Command ${CMAKE_CURRENT_BINARY_DIR}/$<CONFIG>/RunTest.ps1
               ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/test/${name}.py
-              --build_dir=${CMAKE_CURRENT_BINARY_DIR}/$<CONFIG> ${ARGN})
+              --build_dir=${CMAKE_BINARY_DIR}/$<CONFIG>/bin ${ARGN})
         else()
           add_test(NAME ${name}
             COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/test/${name}.py
-              --build_dir=${CMAKE_CURRENT_BINARY_DIR}/$<CONFIG> ${ARGN})
+              --build_dir=${CMAKE_BINARY_DIR}/$<CONFIG>/bin ${ARGN})
         endif()
       else (CMAKE_CONFIGURATION_TYPES)
         # Single-configuration build generators like Makefile generators
@@ -295,11 +295,11 @@ function(py_test name)
           add_test(NAME ${name}
             COMMAND powershell -Command ${CMAKE_CURRENT_BINARY_DIR}/RunTest.ps1
               ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/test/${name}.py
-              --build_dir=${CMAKE_CURRENT_BINARY_DIR} ${ARGN})
+              --build_dir=${CMAKE_BINARY_DIR}/bin ${ARGN})
         else()
           add_test(NAME ${name}
             COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/test/${name}.py
-              --build_dir=${CMAKE_CURRENT_BINARY_DIR} ${ARGN})
+              --build_dir=${CMAKE_BINARY_DIR}/bin ${ARGN})
         endif()
       endif (CMAKE_CONFIGURATION_TYPES)
     else()
@@ -311,11 +311,11 @@ function(py_test name)
         add_test(NAME ${name}
           COMMAND powershell -Command ${CMAKE_CURRENT_BINARY_DIR}/RunTest.ps1
             ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/test/${name}.py
-            --build_dir=${CMAKE_CURRENT_BINARY_DIR}/\${CTEST_CONFIGURATION_TYPE} ${ARGN})
+            --build_dir=${CMAKE_BINARY_DIR}/\${CTEST_CONFIGURATION_TYPE}/bin ${ARGN})
       else()
         add_test(NAME ${name}
           COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/test/${name}.py
-            --build_dir=${CMAKE_CURRENT_BINARY_DIR}/\${CTEST_CONFIGURATION_TYPE} ${ARGN})
+            --build_dir=${CMAKE_BINARY_DIR}/\${CTEST_CONFIGURATION_TYPE}/bin ${ARGN})
       endif()
     endif()
   endif(PYTHONINTERP_FOUND)
